tensorboard/uploader/uploader_main.py

```diff
@@ -439,6 +463,26 @@ def _get_intent(flags):
     raise AssertionError('Unknown subcommand %r' % (cmd,))
 
 
+def _get_server_info(flags):
+  if flags.api_endpoint:
+    return server_info_lib.create_server_info(flags.origin, flags.api_endpoint)
+  else:
+    return server_info_lib.fetch_server_info(flags.origin)
+
+
+def _handle_server_info(info):
+  compat = info.compatibility
+  if compat.verdict == server_info_pb2.VERDICT_WARN:
+    sys.stderr.write('Warning: %s\n' % compat.details)
```

Comment:
Optional, but maybe "Server warning" just to make it a bit clearer in debugging where this is coming from?

Same, also optional, for "Error".

Reply:
Hmm; I like that idea, but “Server error” makes it sound like the server
has an error. We expect these messages usually to be “your client is out
of date”, which is more like a client error than a server error.

What do you think of using “Warning: \[from server\]” instead?

> Warning \[from server\]: Experiencing transient errors; be patient.

> Error \[from server\]: Please upgrade to TensorBoard 3.0.0 or later.